app/services/provider_authorisation.rb

```diff
@@ -71,4 +87,9 @@ def training_provider_can_view_safeguarding_information?(course:)
         .view_safeguarding_information
         .exists?(ratifying_provider: course.accredited_provider, training_provider: course.provider)
   end
+
+  def actor_has_permission_to_make_decisions?(providers:)
+    @actor.is_a?(VendorApiUser) ||
+      @actor.provider_permissions.make_decisions.exists?(provider: providers)
```

Comment:
I think we should assume all potential non-API `@actors` will implement `can_make_decisions_for?` and call that

Reply:
Is this a suggestion to attach `can_make_decisions_for?` methods to `ProviderUser`, 'VendorApiUser` and `SupportUser`? I think I prefer keeping all authorisation logic in this service.

See also https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/2231#discussion_r439458862

Reply:
Although we could add a nicer interface for the query to the actor classes, instead of accessing the one-to-many directly.

Reply:
I've added a `provider.users_with_make_decisions` and removed `can_make_decisions_for?` in https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/2248/commits/288d8c2b50a0256627481db1e3020735910074e5